engine/flink/schemed-kafka-components-utils/src/test/scala/pl/touk/nussknacker/engine/schemedkafka/flink/typeinfo/ConsumerRecordSerializerSpec.scala

```diff
@@ -0,0 +1,118 @@
+package pl.touk.nussknacker.engine.schemedkafka.flink.typeinfo
+
+import org.apache.flink.api.common.ExecutionConfig
+import org.apache.flink.api.common.serialization.SerializerConfigImpl
+import org.apache.flink.api.common.typeutils.TypeSerializer
+import org.apache.flink.api.java.typeutils.TypeExtractor
+import org.apache.flink.core.memory.{DataInputViewStreamWrapper, DataOutputViewStreamWrapper}
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.common.header.internals.{RecordHeader, RecordHeaders}
+import org.apache.kafka.common.record.TimestampType
+import org.scalatest.Assertion
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks
+import pl.touk.nussknacker.engine.flink.test.FlinkTestConfiguration
+import pl.touk.nussknacker.test.ProcessUtils.convertToAnyShouldWrapper
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream}
+import java.nio.charset.StandardCharsets
+import java.time.LocalDate
+import java.util.Optional
+import scala.reflect.{ClassTag, classTag}
+import scala.util.Random
+
+class ConsumerRecordSerializerSpec extends AnyFunSuite with Matchers with TableDrivenPropertyChecks {
```

Comment:
Can we make `SerializableConsumerRecord`  is a POJO and use it in objects passed through DataStreams instead of creating a TypeInfo from the scratch? I see that we have a control on this object - it is created in `KafkaSchemaBasedKeyValueDeserializationSchemaFactory`. Am I right?

Reply:
I can use it, but then API components will be changed, do we want it?

Reply:
And also I think we need rewrite:     headers: Option[Map[String, Option[String]]], to java map.

Reply:
What do you mean by "API components"? The pointed class is in the `kafka-components-utils`. 
With regards to headers - yes, by POJO I mean a deeper changes, for example every field is an `Option` instead of a `@Nullable` field

Reply:
Don't worry, for Option we have dedicated Flink Serializer. I'll prepare some proposition of change.

Reply:
IMO we should try to decouple from scala type info as much as possible

Reply:
As we have decided, we stay with this solution -> special ConsumerRecordTypeInfo.